helix-core/src/main/java/org/apache/helix/InstanceType.java

```diff
@@ -22,7 +22,7 @@
 import java.util.Arrays;
 import java.util.List;
 
-import org.apache.helix.monitoring.mbeans.MonitorDomainNames;
+import org.apache.helix.monitoring.common.mbeans.MonitorDomainNames;
```

Comment:
I think the title of this PR/commit is a bit misleading - are we removing any classes?

I think instead of moving dual-existent classes so that they have different import paths, we should identify which classes both exist in both metrics-common and helix-core, and just remove the ones in helix-core. They were initially left in helix-core, but we should remove them clean since they're causing issues. This way, we won't have to modify any package names and the issue we get from duplicate classes at runtime will no longer be there.

Reply:
Thanks for pointing out the misleading title. Changed it to resolved.

The classes in metrics-common have the same package path org.apache.helix.monitoring as those classes in helix-core. The challenge is, with existing bundle plugin/strategy in helix, there is no way to exclude the metric-common's classes in helix-core.jar.

Even we could change the bundle plugin to support class filtering, I still don't think it is clean if we still put same package classes in different jars. 

The right way for different modules/jars should have different package paths, eg.:
metrics-common: org.apache.helix.monitoring.common
helix-core: org.apache.helix.monitoring

With these different package paths, we could cleanly avoid exporting classes of metrics-common to helix-core.jar.
This is the reason why the PR would like to differentiate the package paths for different modules/jars.

Reply:
Why don't we just remove the conflicting classes from helix-core? That is what my 2nd paragraph is saying. If you just change the package structure for the conflicting classes in metrics-common, you'd be masking the problem (the duality still exists) instead of solving the problem.

Reply:
I guess you missed something about the problem. There are no conflicting classes in helix-core. All classes are unique/only in helix codebase. The problem is, when we package the helix jars, the dependent subproject's classes are packaged into helix-core.jar: all classes in module helix-core and its dependencies match `org.apache.helix.*` are packaged to helix-core.jar. Then there are duplicate classes from metrics-common.jar/zookeeper-api.jar/helix-core.jar

Reply:
So what you're claiming here is when maven packages all of them into one helix-core.jar, that jar ends up having multiple copies of the same class with the same package name?

The description in the issue https://github.com/apache/helix/issues/1198 is still not too clear - could you explain what the failure is exactly?

Reply:
> The problem is, when we package the helix jars, the dependent subproject's classes are packaged into helix-core.jar: all classes in module helix-core and its dependencies match org.apache.helix.* are packaged to helix-core.jar. Then there are duplicate classes from metrics-common.jar/zookeeper-api.jar/helix-core.jar
I believe this is clear. And also I've pasted the failure log message.

Reply:
Thanks for pasting the log message, but this doesn't seem like it's coming from Maven. Is this coming from some other build tool like gradle? Since this detail is missing in the issue/PR description, could you please add this detail?

I think this could be solved by tweaking the ivy file like you did here - do you think we could try to resolve the conflict without changing package names/moving packages?

Reply:
Also, this part was a little misleading:

> And at tests runtime, we have a check to detect duplicate classes and this causes tests failure.

I wanted to check if this test is referring to some test happening outside of the Helix maven project? Could you confirm?